PWGHF/TableProducer/candidateSelectorDstar.cxx

```diff
@@ -0,0 +1,285 @@
+
+  } dstarconfigurable;
+
+  struct CommonConfigurable : ConfigurableGroup {
+    // TPC PID
+    Configurable<double> ptPidTpcMin{"ptPidTpcMin", 0.15, "Lower bound of track pT for TPC PID"};
+    Configurable<double> ptPidTpcMax{"ptPidTpcMax", 5., "Upper bound of track pT for TPC PID"};
+    Configurable<double> nSigmaTpcMax{"nSigmaTpcMax", 3., "Nsigma cut on TPC only"};
+    Configurable<double> nSigmaTpcCombinedMax{"nSigmaTpcCombinedMax", 5., "Nsigma cut on TPC combined with TOF"};
+
+    // TOF PID
+    Configurable<double> ptPidTofMin{"ptPidTofMin", 0.15, "Lower bound of track pT for TOF PID"};
+    Configurable<double> ptPidTofMax{"ptPidTofMax", 5., "Upper bound of track pT for TOF PID"};
+    Configurable<double> nSigmaTofMax{"nSigmaTofMax", 3., "Nsigma cut on TOF only"};
+    Configurable<double> nSigmaTofCombinedMax{"nSigmaTofCombinedMax", 5., "Nsigma cut on TOF combined with TPC"};
+
+    // AND logic for TOF+TPC PID (as in Run2)
+    Configurable<bool> usePidTpcAndTof{"usePidTpcAndTof", false, "Use AND logic for TPC and TOF PID"};
+
+    // CCDB configuration
+    Configurable<std::string> ccdbUrl{"ccdbUrl", "http://alice-ccdb.cern.ch", "url of the ccdb repository"};
+  } commonconfigurable;
+
+  // o2::ccdb::CcdbApi ccdbApi;
+
+  // TrackSelectorPi selectorPion;
+  // TrackSelectorKa selectorKaon;
+
+  using TracksSel = soa::Join<aod::TracksWDcaExtra, aod::TracksPidPi, aod::TracksPidKa>;
+
+  double massPi, massK, massD0;
+  void init(InitContext& initContext)
+  {
+    massPi = o2ServicePDG->Mass(211);
+    massK = o2ServicePDG->Mass(311);
+    massD0 = o2ServicePDG->Mass(421);
+  }
+
+  /// Conjugate-independent topological cuts on D0
+  /// @brief Topological selection on D0 candidate from Dstar
+  /// @tparam T table iterator type of the candidate
+  /// @param candidate candidate instance(object)
+  /// @return true or false depending on selected or not
+  template <typename T>
+  bool selectionTopolD0(const T& candidate)
+  {
+    auto candpT = candidate.d0pt();
+    auto pTBin = findBin(d0configurable.binsPtForD0, candpT);
+    if (pTBin == -1) {
+      return false;
+    }
+
+    // check that the candidate pT is within the analysis range
+    if (candpT < d0configurable.ptD0CandMin || candpT >= d0configurable.ptD0CandMax) {
+      return false;
+    }
+    // product of daughter impact parameters
+    if (candidate.d0impactParameterProduct() > d0configurable.cutsForD0->get(pTBin, "d0d0")) {
+      return false;
+    }
+    // // cosine of pointing angle
+    if (candidate.d0cpaXY() < d0configurable.cutsForD0->get(pTBin, "cos pointing angle xy")) {
+      return false;
+    }
+    // normalised decay length in XY plane
+    if (candidate.decayLengthXYNormalised() < d0configurable.cutsForD0->get(pTBin, "normalized decay length XY")) {
+      return false;
+    }
+    // chi2PCA
+    // if(candidate.chi2PCA() > d0configurable.cutsForD0->get(pTBin,"DCA")){
+    //   return false;
+    // }
```

Comment:
Why is this not used?

Reply:
This was also not used in D0 selector task because they did not define cut for chi2PCA in SelectionCuts.h. Now I have defined this cut in SelectionCuts.h under namespace:hf_cuts_dstar_to_pi_d0  so that i did not touch the code of D0.